### PR TITLE
Fix rlient issue

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. Fix segment fault if R_init() is failed.
2. Fix librcall.so not found issue, if the current working
   directory is not the librcall.so path
3. Refactor code to match upstream and pyclient

### CODE REVIEW

File: rcall.c [hunk 1]
```diff
@@ -145,37 +188,47 @@ void r_init( ) {
...
-    load_r_cmd(OPTIONS_NULL_CMD);
+    status = load_r_cmd(cmds[0]);
+
+    if (status < 0) {
+        return -1;
+    }
 
     /* next load the plr library into R */
-    load_r_cmd(buf=get_load_self_ref_cmd("librcall.so"));
+    if (readlink("/proc/self/exe", path, PATH_MAX) == -1) {
```
Comment (paul-guo-) [inline rcall.c]: Add Linux guard?

**Comment by paul-guo-** [comment]
Maybe a more accurate title ?

File: rcall.c [hunk 2]
```diff
@@ -145,37 +188,47 @@ void r_init( ) {
...
+        lprintf(ERROR, "can not read execute path");
+    }
+
+    if((p = strrchr(path, '/'))) {
+        *(p+1) = '\0';
+    }
+
+    status = load_r_cmd(buf=get_load_self_ref_cmd(path));
```
[the remark this answers is not in the document]
Comment (HaozhouWang) [reply rcall.c]: Sure, I moved this into get_load_self_ref_cmd()

File: rcall.c [hunk 3]
```diff
@@ -145,37 +188,47 @@ void r_init( ) {
...
-    load_r_cmd(PG_LOG_WARNING_CMD);
-    load_r_cmd(PG_LOG_ERROR_CMD);
-    load_r_cmd(PG_LOG_FATAL_CMD);
-    load_r_cmd(SPI_DBGETQUERY_CMD);
+    if (status < 0) {
+        return -1;
+    }
+
+    for (i = 1; (cmd = cmds[i]); i++) {
+        status = load_r_cmd(cmds[i]);
```
[the remark this answers is not in the document]
Comment (HaozhouWang) [reply rcall.c]: Thanks, add to print the error message in plc_raise_delayed_error()

File: rcall.c (hunk 1 above)
Comment (HaozhouWang) [reply rcall.c]: Sure, add __linux__

**Pull request merged**